### ISSUE TITLE
Sort rows by some other deterministic field when primary sort fields are identical

### ISSUE THREAD

**Issue by PaulMcF1987** [Budibase/budibase#6732]
Since it is not currently possible to filter via `relationship columns` I have a separate entry box where I type the customers name, after selecting it from the relationship dropdown,

From there, I can give my customers access to the app to track their jobs using the define filters option and setting condition to change the filter for each customer.

I have found though that the filter appears to be temperamental and misses out some records.

For example, my customer called me as they could not see a job on the portal, the job could have been at the bottom of page one or the top of page 2.

When I scrolled down to the bottom of page 1 in the table it was not there, when I went over to page 2, where it should have been, the top jobs were the same as the bottom jobs on page 1... I have added pics below to show.

![image](https://user-images.githubusercontent.com/75368133/179217877-80eafdba-bad2-4f4b-8c86-5f0bac4884b1.png)

The above image shows page 1.

Below shows page 2. there are already some pre-ticked rows and you will notice that the `PoNo` is the same as jobs from the bottom of page 1.

![image](https://user-images.githubusercontent.com/75368133/179218861-3b982e52-bfd1-40a2-b3ba-305947012ccd.png)

There has also been a sort order used - `JobDate - Descending`

I also tried changing the filter to `fk_customer_jobrecord` and used the relevant number for each customer but it gave the exact same output.

When I changed the `order by` terms and searched, the jobs in question were shown so there seems to be an error with the ordering.

**Comment by PaulMcF1987** [comment]
@aptkingston 
Yeah the ordering works fine, I have attached the same 2 pics with the dates showing, you will see that they descend in the expected order.

If I change the table to show 20 records rather than 10 it shows all the expected records

The error only seems to happen if the date at the bottom of one page is the same as the date at the top of the next page, if the date on the next page is different, the error doesnt seem to happen.

![image](https://user-images.githubusercontent.com/75368133/179226836-5eb72426-971a-49b2-a5c7-7d6dfca3c909.png)


![image](https://user-images.githubusercontent.com/75368133/179226623-617d3949-5fb6-48a2-a53b-c8dd29aabe95.png)

**Comment by aptkingston** [comment]
Perfect thanks for the pictures.Yea it looks like since they have the same date then there is randomness in their order. We could fall back to ordering by something internal (like their internal Budibase ID) as second precedence when their primary sort fields are identical. I'll update the title this issue to reflect the work that needs done :+1:

**Comment by PaulMcF1987** [comment]
@aptkingston 
One thing that might make a difference to the way you approach it is, I am using an external MYSQL DB, I have not used the internal CouchDB for this
